### ISSUE TITLE
[3.x] - The "No sniffs were registered." is outputted to STDOUT instead of STDERR.

### ISSUE THREAD

**Issue by yCodeTech** [PHPCSStandards/PHP_CodeSniffer#1283]
## Describe the bug

In the latest of 3.x (3.13.4), the `"No sniffs were registered"` error (and maybe other errors?) seems to be outputted to `STDOUT` for both phpcs and phpcbf. Shouldn't this be to `STDERR`?

I'm now a co-maintainer of the [vscode extension PHP Sniffer & Beautifier](https://github.com/valeryan/vscode-phpsab), and while the 3.x is still supported, trying to catch these errors in `STDOUT` for the phpcbf is particularly painful, while also getting the potentially fixed content in other scenarios. 

In 4.x, this particular error is outputted to `STDERR`, so would it be possible to backport this into 3.x please`

### Custom ruleset
To test the "No sniffs were registered" error, I'm using an empty ruleset.
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <description>If you are using a custom ruleset, please enter the relevant part here.</description>
</ruleset>
```

## Expected behavior

The `"No sniffs were registered"` and other errors to be sent to `STDERR`.

## Versions (please complete the following information)

|                         |                         |
|-------------------------|-------------------------|
| Operating System        | Windows 10              |
| PHP version             | 8.2.9                   |
| PHP_CodeSniffer version | 3.13.4                  |
| Standard                | None                    |
| Install type            | Composer (global/local) |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [ ] I have verified the issue still exists in the `4.x` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@yCodeTech In PHPCS 3.x and earlier, there was no separation of output at all and everything always went to STDOUT.
This has now been changed in PHPCS 4.0 via https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/1010 to fix https://github.com/squizlabs/PHP_CodeSniffer/issues/1612.

Backporting this is not on the table as the change in output streams is a breaking change and therefore belongs in a major.

You may want to have a look at the `--report-file` option to see if that can be useful to you.

> while the 3.x is still supported

3.x is no longer supported, safe for security and PHP runtime compatibility fixes now 4.0 has been released.
